appservice/src/createAppService/SiteCreateStep.ts

```diff
@@ -87,17 +97,16 @@ export class SiteCreateStep extends AzureWizardExecuteStep<IAppServiceWizardCont
                 storageConnectionString = `DefaultEndpointsProtocol=https;AccountName=${storageAccount.name};AccountKey=${keysResult.keys[0].value};EndpointSuffix=${endpointSuffix}`;
             }
 
-            if (this.createFunctionAppSettings) {
-                newSiteConfig.appSettings = await this.createFunctionAppSettings({
-                    storageConnectionString,
-                    fileShareName,
-                    // tslint:disable-next-line:no-non-null-assertion
-                    os: nonNullProp(wizardContext, 'newSiteOS'),
-                    // tslint:disable-next-line:no-non-null-assertion
-                    runtime: nonNullProp(wizardContext, 'newSiteRuntime'),
-                    // tslint:disable-next-line: strict-boolean-expressions
-                    aiInstrumentationKey: wizardContext.appInsightsComponent && wizardContext.appInsightsComponent ? wizardContext.appInsightsComponent.instrumentationKey : undefined
-                });
+            if (this.createSiteAppSettings) {
```

Comment:
Move this out of the if/else so that you have less duplicate code. Something like this:
```
let storageConnectionString: string | undefined;
let fileShareName: string | undefined;
if (wizardContext.newSiteKind === AppKind.app) {
    ...
} else {
    ...
}

if (this.createSiteAppSettings) {
    newSiteConfig.appSettings = await this.createSiteAppSettings(
    {
        ...
    }
}
```